ikarus/finiteelements/mechanics/nonlinearelastic.hh

```diff
@@ -264,6 +271,26 @@ protected:
     }
   }
 
+  template <typename ScalarType>
+  void calculateMassImpl(const Requirement& par, const MatrixAffordance& affordance,
+                         typename Traits::template MatrixType<> M) const {
+    for (const auto& [gpIndex, gp] : localBasis_.viewOverIntegrationPoints()) {
+      const auto intElement = geo_->integrationElement(gp.position()) * gp.weight();
+      auto& N               = localBasis_.evaluateFunction(gpIndex);
+
+      auto nopI = Eigen::Matrix<double, myDim, myDim>::Zero().eval();
+      auto nopJ = Eigen::Matrix<double, myDim, myDim>::Zero().eval();
+
+      for (size_t i = 0; i < numberOfNodes_; ++i) {
+        nopI.diagonal().setConstant(N[i]);
+        for (size_t j = 0; j < numberOfNodes_; ++j) {
+          nopJ.diagonal().setConstant(N[j]);
+          M.template block<myDim, myDim>(i * myDim, j * myDim) += nopI.transpose() * density_ * nopJ * intElement;
```

Comment:
Same here, these trival mass matrix cases can also be wrapped into a common function